gnd/src/main/java/com/google/android/gnd/rx/ValueOrError.java

```diff
@@ -72,4 +62,38 @@ public String toString() {
   public static <T> Observable<T> ignoreErrors(Observable<ValueOrError<T>> observable) {
     return observable.filter(voe -> voe.value().isPresent()).map(voe -> voe.value().get());
   }
+
+  public Optional<T> value() {
+    return Optional.ofNullable(value);
+  }
+
+  public Optional<Throwable> error() {
+    if (error != null) {
+      Timber.e(error);
+    }
+    return Optional.ofNullable(error);
+  }
+
+  @NonNull
+  @Override
+  public String toString() {
+    return error().map(t -> "Error: " + t).orElse("Value: " + value);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof ValueOrError)) {
+      return false;
+    }
+    ValueOrError<?> that = (ValueOrError<?>) o;
+    return Objects.equals(value, that.value) && Objects.equals(error, that.error);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(value, error);
+  }
```

Comment:
What are equals() and hasCode() ever used? If not, please remove.

Reply:
Same as above, used in unit test